biocrnpyler/dna_assembly_promoter.py

```diff
@@ -125,25 +127,31 @@ def update_species(self, **keywords):
         mech_tx = self.mechanisms["transcription"]
         
         species = [] #A list of species must be returned
-        species += mech_tx.update_species(dna = self.assembly.dna, transcript = self.transcript, regulator = self.activator, part_id = self.name+"_"+self.activator.name, component = self)
+        species += mech_tx.update_species(dna = self.assembly.dna, transcript = self.transcript, regulator = self.activator, part_id = self.name+"_"+self.activator.name, leak = self.leak, component = self)
         
         return species
 
     def update_reactions(self, **keywords):
         mech_tx = self.mechanisms["transcription"]
         
         reactions = [] #a list of reactions must be returned
+
+
         reactions += mech_tx.update_reactions(dna = self.assembly.dna, transcript = self.transcript, regulator = self.activator, 
-                                             component = self, part_id = self.name+"_"+self.activator.name, **keywords)
+                                             component = self, part_id = self.name+"_"+self.activator.name, leak = self.leak, **keywords)
+
+        
         return reactions
 
 #A class for a promoter which can be repressed by a single species, modelled as a negative hill function
-class RepressablePromotor(Promoter):
-    def __init__(self, name, repressor, transcript = None, **keywords):
+class RepressablePromoter(Promoter):
```

Comment:
Sorry for being a grammar nazi, but I think it is called a RepressiblePromoter
e.g. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3599224/

Reply:
Sorry for being a bad speller. Typo-corrected.

Reply:
no worries, just this PR it the perfect place to weed out this typo.  Thanks for taking care of it!